metrics.go

```diff
@@ -67,15 +60,10 @@ func (t *counter) Emit() {
 		return
 	}
 
-	go func(key string, count int) {
-		select {
-		case statQueue <- NewCounter(
-			fmt.Sprintf("%s.%s", metricPrefix, key),
-			count,
-		):
-		default:
-		}
-	}(t.key, t.count)
+	statQueue <- NewCounter(
```

Comment:
Is this method thread safe? If not, this might block other threads from calling emit on the same counter, if the counter object is being shared by other threads.